### ISSUE TITLE
[1.19.3][1.201] Keykinds get overwritten every single pack load, not only when they are missing.

### ISSUE THREAD

**Issue by ProsperCraft** [blay09/DefaultKeyConfig#70]
### Minecraft Version

1.19.3 and 1.20.1

### Mod Loader

Forge

### Mod Loader Version

44.1.21

### Mod Version

defaultoptions-forge-1.19.3-16.0.1.jar and defaultoptions-forge-1.20-18.0.0.j

### Describe the Issue

Options.txt is being overwritten every single pack load, not only when it is missing, resetting players changes.

### Logs

https://gist.github.com/ProsperCraft/bc346e2aedf373a9448b2fbaaae33c71

**Comment by ClaudiusMinimus** [comment]
Also happens in defaultoptions-forge-1.19-15.0.1.jar if default configs are in `/minecraft/config/defaultoptions/extra/config/`

**Comment by BlayTheNinth** [comment]
I cannot reproduce either issue running the respective latest versions of only Balm and Default Options. Do you have steps to reproduce or an example pack where it is broken?

@ClaudiusMinimus The file was not marked correctly as 1.19.2 compatible on CurseForge, causing the launcher to downgrade it back to 15.0.0 if the update button was used. Could that be what happened?

**Comment by ClaudiusMinimus** [comment]
> I cannot reproduce either issue running the respective latest versions of only Balm and Default Options. Do you have steps to reproduce or an example pack where it is broken?

You can install/download this: https://www.curseforge.com/minecraft/modpacks/create-on-a-potato-pc-sky/files/4431009
> 
> @ClaudiusMinimus The file was not marked correctly as 1.19.2 compatible on CurseForge, causing the launcher to downgrade it back to 15.0.0 if the update button was used. Could that be what happened?

I use GDLauncher to pull in mod updates and to build the CF install files. I am running Linux, so the CF launcher or Install button does not work on my system.

**Comment by BlayTheNinth** [comment]
That pack does indeed reference and download the older 15.0.0 version that did not include the fix yet.

from manifest.json: `{"projectID":232131,"fileID":3832590,"required":true}`
(15.0.0 is fileID `3832590`, 15.0.1 is fileID `4412620`)

Maybe CurseForge automatically downgraded it to 15.0.0 on upload when it saw that 15.0.1 was not marked as 1.19.2 compatible yet.

**Comment by ClaudiusMinimus** [comment]
Thanks! Will test it on my next update.
Cheers!
CM

**Comment by ProsperCraft** [comment]
We are having the issue with defaultoptions-forge-1.19.3-16.0.1.jar as indicated, and I don't see any other version to download.

**Comment by BlayTheNinth** [comment]
@ProsperCraft Does the issue stop happening with Default Options removed or with only Default Options installed? I cannot reproduce your issue and my best guess would be that it's something conflicting with another mod in the pack.

**Comment by ProsperCraft** [comment]
Yes the issue stops when you disable Default Options.

**Comment by BlayTheNinth** [comment]
Just tested it again with the exact same versions and everything worked fine.

Example Pack:
[DefaultOptions-70.zip](https://github.com/TwelveIterationMods/DefaultOptions/files/11006025/DefaultOptions-70.zip)

Configured Defaults: 
- Jump Button to Key `r`
- FOV: Quake Pro
- View Bobbing: Off

Changing any of the settings or the key and restarting Minecraft keeps my changes and does not reset back to the defaults.

Unless I get more information there's nothing I can do.

**Comment by Xzeni** [comment]
Having the same issue with 1.19.4 (defaultoptions-forge-1.19.4-17.0.1.jar) with just the keybinds. Works fine on a fresh install, but gets reset every time Minecraft relaunches. Weird thing is that it doesn't even reset to the default I set it to, but the mod default. I will try to experiment with it soon and send you results.

Modpack: [Xzeni's SMP](https://www.curseforge.com/minecraft/modpacks/xzenis-smp)
How to reproduce: Change keybinds and relaunch Minecraft.

**Comment by TheIllusiveC4** [comment]
Minecraft 1.20.1 - Forge 47.0.6
Default Options 18.0.0
Balm 7.0.4

I've also encountered this issue. Something that hasn't been mentioned so far in this thread is that it notably only seems to affect modded keybindings/options. Vanilla ones act as expected.

**Steps to reproduce**
1) Along with the above versions, install Curios 5.2.0-beta.3+1.20.1 or Just Enough Items 15.1.0.19 (these are just two examples of mods that add keybindings and the ones I tested with, any other mod could likely substitute).
2) Change a vanilla keybinding.
3) Change a modded keybinding.
4) Restart Minecraft and the modded keybinding will have reset to the mod's default while the vanilla keybinding remains as set in step 2.
